ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java

```diff
@@ -201,11 +203,37 @@ private Response makeGetRequest(String path) {
     public QueryStream(Response response) {
       this.response = response;
 
+      this.closed = false;
```

Comment:
`closed` should be volatile if it is read/updated from multiple threads unless it is always surrounded with `synchronized(...)`